aptihramy-backend/database/database.py

```diff
@@ -220,3 +234,21 @@ def get_tracked_features(self) -> tuple[list[str], list[str]]:
         raw = [feature.name for feature in self._record_schema.fields]
         pretty = [COLUMN_RAW_TO_PRETTY[col] for col in raw]
         return (raw, pretty)
+
+    def get_record(self, frame_idx: int, record_idx: int) -> pl.DataFrame:
+        if not (0 <= frame_idx < len(self._dataframes)):
+            raise IndexError(f"Frame index {frame_idx} is out of range.")
+
+        frame = self._dataframes[frame_idx]
+
+        if not (0 <= record_idx < len(frame)):
+            raise IndexError(
+                f"Record index {record_idx} is out of range for frame {frame_idx}."
+            )
+        return frame[record_idx]
+
+    def get_tracked_years(self) -> list[int]:
+        return self._tracked_years
+
+    def get_tracking_chain(self, tracker_id: ID):
```

Comment:
Here you should probably use `self._graph.materialize_tracking_chain` and not use an internal function directly.
Also you should indicate the returning type of the function.